### PR TITLE
[Enhancement] Implement format methods for Status

### LINKED ISSUE #25974: [Enhancement] Implement format methods for `Status`
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Implementing format methods for `Status`, then developers can format status directly  in `fmt`.
Current:
```c++
fmt::format("{}", status.to_string());
```
Expected:
```c++
fmt::format("{}", status);
```

### Solution

See https://wgml.pl/blog/formatting-user-defined-types-fmt.html

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Dragonliu2018**
## Proposed changes

Issue Number: close #25974 

Implement format methods for Status.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/test/common/status_test.cpp [hunk 1]
```diff
@@ -73,4 +73,10 @@ TEST_F(StatusTest, Error) {
     }
 }
 
+TEST_F(StatusTest /*unused*/, Format /*unused*/) {
+    // default
+    Status st;
```
Comment (yiguolei) [inline be/test/common/status_test.cpp:78]: please add 3 tests：
1.  status == ok
2.  status == error && has stacktrace
3. status == error && do not have stacktrace

Comment (Dragonliu2018) [reply be/test/common/status_test.cpp:78]: Whether status contains stacktrace is determined by the preprocessor macro `ENABLE_STACKTRACE`. So for cases 2 and 3, I just need to test `status == error`, not care stacktrace?

Comment (Dragonliu2018) [reply be/test/common/status_test.cpp:78]: if not, please give me an example.

**Review by platoneko — APPROVED** [review]

**Comment by platoneko** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**